Shared/sdk/SharedUtil.Crypto.h

```diff
@@ -25,18 +26,66 @@ namespace SharedUtil
         SString publicKey, privateKey;
     };
 
-    inline SString Base64encode(const SString& data)
+    inline SString Base64encode(const SString& data, const SString& variant = SString())
     {
-        SString                result;
-        CryptoPP::StringSource ss(data, true, new CryptoPP::Base64Encoder(new CryptoPP::StringSink(result), false));            // Memory is freed automatically
+        SString result;
+
+        if (variant == "URL")
+        {
+            CryptoPP::StringSource ss(data, true, new CryptoPP::Base64URLEncoder(new CryptoPP::StringSink(result), false));
+        }
+        else
+        {
+            CryptoPP::StringSource ss(data, true, new CryptoPP::Base64Encoder(new CryptoPP::StringSink(result), false));
+        }
+
+        return result;
+    }
+
+    inline SString Base64decode(const SString& data, const SString& variant = SString())
+    {
+        std::string result;
+
+        if (variant == "URL")
+        {
+            CryptoPP::StringSource ss(data, true, new CryptoPP::Base64URLDecoder(new CryptoPP::StringSink(result)));
+        }
+        else
+        {
+            CryptoPP::StringSource ss(data, true, new CryptoPP::Base64Decoder(new CryptoPP::StringSink(result)));
+        }
+
+        return result;
+    }
+
+    inline SString Base32encode(const SString& data, const SString& variant = SString())
+    {
+        std::string result;
+
+        if (variant == "HEX")
```

Comment:
This is debatable but I would prefer a enum here.

Reply:
I know that this will require a conversion from string to enum. I'm just pointing out the possible improvements. Due to a small string length the current implementation is not that bad actually.

Reply:
Since base64 and base32 have separate variants (base64 - url, base32 - hex) and there are no of others, they should be hardcoded if we don't want to create two separate enum classes, which doesn't make any sense, the variants wouldn't be expandable anyway, because they don't exist. If this is not enough reason to keep it the current way, please write a comment in the next review.